WalletWasabi/Crypto/StrobeProtocol/StrobeHasher.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+using NBitcoin;
+
+namespace WalletWasabi.Crypto.StrobeProtocol
+{
+	public sealed class StrobeHasher
+	{
+		private Strobe128 strobe;
+
+		private StrobeHasher()
+		{
+			strobe = new Strobe128("WabiSabi_v1.0");
+			Append("domain-separator", "round-parameters");
+		}
+
+
+		public void Append<T>(string fieldName, T fieldValue) where T : notnull
+		{
+			strobe.AddAssociatedMetaData(Encoding.UTF8.GetBytes(fieldName), false);
+
+			var serializedValue = Encoding.UTF8.GetBytes($"{fieldValue}");
```

Comment:
Since `Append` is already parameterized by `T`, maybe we should require it to be an interface that can convert to bytes? I don't like the idea of relying on `$"{}"` and then encoding the resulting string as UTF-8 because that depends on the default stringification, especially for values that already have a clear byte representation.

Reply:
There is no common interface to `uint`, `ulong`, `Money`, `CredentialIssuerPrarameters`, etc. I have improved the solution to avoid trusting on the default stringification because it is rather dangerous. I also made sure to enforce `invariant culture` so the string representation of a `decimal` for example doesn't change depending on the computer's localization.

Reply:
I like the revised case switch approach better:

1. closed world, full set of supported types is self evident in one chunk of code
2. byte level encoding semantics are complete and unambiguous

However, it still canonicalizes to strings which are implicitly utf8, this doesn't really make sense for binary data without also assuming some encoding.

We can simplify it further by using overloading instead of a switch, where the basic `Append` (private) only supports `[]byte`, a private overload that supports `string` encodes as UTF8, and the public API supports a specific list of types. Thoughts?

Reply:
Yep. Done.